tiledb/sm/config/config.cc

```diff
@@ -527,6 +527,25 @@ Status Config::get(const std::string& param, T* value, bool* found) const {
   return Status::Ok();
 }
 
+template <class T>
+T Config::get(
+    const std::string& key,
+    [[maybe_unused]] const Config::MustFindMarker& marker) const {
+  bool found;
+  const char* val = get_from_config_or_env(key, &found);
+  if (!found) {
+    throw_config_exception("Failed to get config value for key: " + key);
+  }
+  T converted_value;
+  auto status = utils::parse::convert(val, &converted_value);
+  if (!status.ok()) {
+    throw_config_exception(
+        "Failed to convert configuration value '" + std::string(val) +
```

Comment:
We should avoid copypasta. We can use the same, internal implementation function for both new and old functions. It would need a `template<bool>` argument (or similar) to pick `must_find` or `need_not_find` behavior.